### ISSUE TITLE
Publish to NuGet

### ISSUE THREAD

**Issue by slorion** [RxDave/Rxx#1]
Would you mind publishing an updated version of Rxx to NuGet ? That would save us the hassle of updating/compiling/publishing Rxx to our internal nuget server. Thank you in advance !

**Comment by RxDave** [comment]
Yes, people have been asking this for a while. I'll have to make time for it, thanks for the (eighth?) reminder :)

**Comment by slorion** [comment]
Ok thank you, I know first hand that maintaining a popular open source project can be demanding :)

**Comment by danbarua** [comment]
@slorion Would you mind sharing how you got Rxx to build? I'm struggling here!

**Comment by slorion** [comment]
We are not using this version, but an earlier one from CodePlex that we got a while ago. It was still targeting an old version of Rx.NET, so we have integrated the source code in our solution (only for .NET 4.5) and then fixed some resulting compile errors when updating it to Rx.NET 2.2.5.

**Comment by danbarua** [comment]
Cool, thanks. I was considering that approach. :+1:

**Comment by NickRedwood** [comment]
As I'm sure you know the RX namespaces have changed in version 3.0.0. Are there any plans to update this project to use those? Thanks for the great library btw.

**Comment by RxDave** [comment]
Rxx hasn't been updated in a while. Every time I think I will, I notice that there's something else more useful and interesting for me to do instead. I haven't actually used Rxx in a few years either, so for a time I was just maintaining it for others to use, and that's pretty boring. I guess I'm worried about falling into the rabbit hole, and priorities.

I find that when I really need a feature from Rxx, it's pretty quick to just throw it together again, often slightly different, and better. Furthermore, I'm currently working for a company with a strict no-sharing policy, so unfortunately I can't even contribute back to my own project when I solve useful problems at the office :|

It's becoming less and less important to me to maintain Rxx. That being said, I could see cleaning it up a lot and using it again in the future. It's still on the table. But I probably won't get around to it until at least I've reached some of my bigger goals for Qactive, which is far more unique, interesting and useful to me than Rxx at the moment.

Sorry for the rant, and thanks for your interest and your comment!  :)

**Comment by NickRedwood** [comment]
Can definitely understand the desire to pursue more interesting work! I can only imagine how much time just the versioning, supporting different frameworks and packaging (inc. Nuget) would take. The annoying thing with the RX 3.0.0 namespace change is that it appears all the old Rx-Main Nuget packages are now unavailable apart from 1.0. So when I installed Rxx from Nuget it installed Rx-Main 1.0 at the same time and that gave a conflict with the System.Reactive 3.0 packages. Erk. Anyway, fixed that now by cloning, changing references and rebuilding myself.

I found rxx through wanting a streamreader and filesystemwatcher to output iobservables, and I'm sure that sort of code will continue to be relevant. It's very useful to me anyway. I'll have to have a look at Qactive at some stage (beyond reading the blurb) - it sounds interesting.

**Comment by danbarua** [comment]
> Furthermore, I'm currently working for a company with a strict no-sharing policy, so unfortunately I can't even contribute back to my own project when I solve useful problems at the office

This makes me sad.

Thanks for everything you've done so far.

> I find that when I really need a feature from Rxx, it's pretty quick to just throw it together again, often slightly different, and better.

This is true. I think the value in Rxx is a repository of patterns and examples that you can paste into your own code rather than bringing in the entire library now.

**Comment by auxon** [comment]
Dave I want to port and maintain Qactive at least if not Rxx parsers.

**Comment by auxon** [comment]
Dave you know I've been working with Rx.NET forever, well, I use the Rxx parsers from time to time and the QActive library too.
